handler/aws.go

```diff
@@ -27,6 +27,15 @@ func init() {
 		host := fmt.Sprintf("execute-api.%s.amazonaws.com", region)
 		services[host] = endpoints.ResolvedEndpoint{URL: fmt.Sprintf("https://%s", host), SigningMethod: "v4", SigningRegion: region, SigningName: "execute-api"}
 	}
+
+	// Add s3 endpoints
+	for region := range endpoints.AwsPartition().Regions() {
+		formats := []string{"s3-%s.amazonaws.com", "s3.%s.amazonaws.com"}
```

Comment:
The format `"s3.%s.amazonaws.com"` is taken care of in the first part of this `init`.  There are a couple of other supported formats on https://docs.aws.amazon.com/general/latest/gr/rande.html for s3 - I'd hesitate to add support for all of them unless there is a specific reason you can't use the current form.  (I'm not completely against it - just less lines of code to maintain 😄 )

Reply:
oh, my mistake, 
i first tried "<bucket>.s3-<region>.amazonaws.com" and got failed, 
and found the example only mentioned "s3.amazonaws.com", 
then I thought neither "s3.<region>.amazonaws.com" or "s3-<region>.amazonaws.com" are supported...
it may turn out to be documentation issue..so I may close this pr..